build.gradle

```diff
@@ -1,46 +1,202 @@
+import org.grails.gradle.plugin.doc.PublishGuideTask
+
+buildscript {
+    repositories {
+        maven { url "https://repo.grails.org/grails/core" }
+        mavenCentral()
+        maven {
+            url = 'https://repo.gradle.org/gradle/libs-releases'
+            description = 'Needed for Gradle Tooling API'
+        }
+    }
+    dependencies {
+        classpath "org.grails.grails-doc:org.grails.grails-doc.gradle.plugin:$grailsVersion"
+        classpath "org.grails:grails-gradle-plugin:$grailsVersion"
+        classpath "org.gradle:gradle-tooling-api:$gradleToolingApiVersion"
+    }
+}
+
 plugins {
-    id 'groovy'
-    id 'java-library'
-    id 'io.github.gradle-nexus.publish-plugin'
-    id 'maven-publish'
+    id "java-library"
+    id "eclipse"
+    id "idea"
     id 'signing'
+    id 'maven-publish'
 }
 
+apply plugin: "org.grails.grails-plugin"
+apply plugin: 'org.grails.grails-doc'
+apply plugin: "io.github.gradle-nexus.publish-plugin"
+
 version =  projectVersion
 group = 'org.grails.plugins'
-ext.set('grailsVersion', libs.versions.grails.get())
-ext.set('isReleaseVersion', !version.toString().endsWith('-SNAPSHOT'))
-ext.set('isSnapshot', !isReleaseVersion)
 
-apply plugin: 'org.grails.grails-plugin' // Needs to be applied after grailsVersion has been set
+
+ext.set('isSnapshot', projectVersion.endsWith('-SNAPSHOT'))
+ext.set('isReleaseVersion', !isSnapshot)
+
+ext.set('signing.keyId', project.findProperty('signing.keyId') ?: System.getenv('SIGNING_KEY'))
+ext.set('signing.secretKeyRingFile', project.findProperty('signing.secretKeyRingFile') ?:  System.getenv('SIGNING_PASSPHRASE') ?: "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg")
+ext.set('signing.password', project.findProperty("signing.password") ?: System.getenv('SIGNING_PASSPHRASE'))
 
 repositories {
   mavenLocal()
   mavenCentral()
   maven { url = 'https://repo.grails.org/grails/core' }
 }
 
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(17)
+    }
+}
+
+dependencyManagement {
+    imports {
+        mavenBom "org.grails:grails-bom:$grailsVersion"
+    }
+    applyMavenExclusions false
+}
+repositories {
+    mavenCentral()
+    maven { url "https://repo.grails.org/grails/core" }
+}
+
+dependencies {
+
+    api "org.quartz-scheduler:quartz:$quartzVersion"
+
+    implementation "org.grails:grails-bootstrap:$grailsVersion"
+    implementation "org.grails:grails-core:$grailsVersion"
+    implementation "org.apache.groovy:groovy-sql:$groovyVersion"
+    implementation "org.springframework:spring-beans:${springVersion}"
```

Comment:
Since the grails-bom extends the spring bom, the specific versions for springVersion, core, bootstrap, & groovy shouldn't be needed.

Reply:
Fixed